tasks/main.yml

```diff
@@ -1,52 +1,93 @@
+    state: started
+    detach: true
+    recreate: yes
+    restart_policy: always
+    published_ports:
+      - "{{ host_port }}:9000"
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /opt/portainer:/data
+
+- name: Wait for container
+  wait_for:
+    port: "{{ host_port }}"
+    host: "{{ inventory_hostname }}"
+
+- name: Configure admin user password
   uri:
-    url: "http://{{ ansible_default_ipv4.address }}:5000/api/users/admin/init"
+    url: "{{ portainer_endpoint }}/users/admin/init"
     method: POST
     return_content: yes
-    body: {"password":"{{ portainer_pass }}"}
     body_format: json
+    body: { "Username":"{{ admin_user }}", "Password":"{{ admin_password }}"}
+  when: admin_user and admin_password is defined
 
-- name: Authenticate against the API (for cluster)
+- name: Generate authentication token
   uri:
-    url: "http://{{ ansible_default_ipv4.address }}:{{ portainer_port }}/api/auth"
+    url: "{{ portainer_endpoint }}/auth"
     method: POST
     return_content: yes
-    body: {"username": "admin", "password": "{{ portainer_pass }}"}
     body_format: json
+    body: {"Username": "{{ admin_user }}", "Password": "{{ admin_password }}"}
   register: auth_token
-  when: mode == "cluster"
-
-- debug: msg="{{ (auth_token.content|from_json).jwt }}"
-  when: mode == "cluster"
+  when: admin_user and admin_password is defined
 
-- name: Define endpoints
+- name: Define Endpoints
   uri:
-    url: "http://{{ ansible_default_ipv4.address }}:{{ portainer_port }}/api/endpoints"
+    url: "{{ portainer_endpoint }}/endpoints"
     method: POST
     return_content: yes
     headers:
       Authorization: "{{ (auth_token.content|from_json).jwt }}"
-    body: {"Name":"{{ hostvars[item]['ansible_host'] }}", "URL":"tcp://{{ hostvars[item]['ansible_host'] }}:2375"}
+    body: {"Name":"{{ item.name }}", "URL":"{{ item.url }}"}
     body_format: json
   with_items:
-    - "{{ groups['{{ swarm_nodes_group }}'] | difference(groups['{{ swarm_managers_group }}']) }}"
-  when: mode == "cluster"
+    - "{{ endpoints | list }}"
+  when: endpoints is defined
+
+- name: Configure Portainer settings
+  uri:
+    url: "{{ portainer_endpoint }}/settings"
+    method: PUT
+    return_content: yes
+    headers:
+      Authorization: "{{ (auth_token.content|from_json).jwt }}"
+    body_format: json
+    body: "{{ lookup('template','settings.json.j2') }}"
+  when: portainer_settings
 
-- name: Add remote endpoints
+- name: Configure Registry
   uri:
-    url: "{{ portainer_endpoint }}"
+    url: "{{ portainer_endpoint }}/registries"
     method: POST
     return_content: yes
     headers:
       Authorization: "{{ (auth_token.content|from_json).jwt }}"
     body_format: json
-    body: {"Name":"{{ remote_endpoint_name }}", "URL":"tcp://{{ remote_endpoint_ip }}:{{ remote_endpoint_port }}"}
-  when: remote_endpoint is defined
+    body: "{{ lookup('template','registry.json.j2') }}"
+  when: docker_registery
```

Comment:
Again - this should be `false` by default

Reply:
fixed